app/src/js/common/session_init.tsx

```diff
@@ -249,24 +249,36 @@ function loadPointClouds (): void {
   for (const item of items) {
     const pcImageMap: {[id: number]: THREE.BufferGeometry} = {}
     Session.pointClouds.push(pcImageMap)
+    const attemptsMap: {[id: number]: number} = {}
     for (const key of Object.keys(item.urls)) {
       const sensorId = Number(key)
       if (sensorId in state.task.sensors &&
           state.task.sensors[sensorId].type === DataType.POINT_CLOUD) {
         const url = item.urls[sensorId]
-        loader.load(
-          url,
-          (geometry: THREE.BufferGeometry) => {
-            Session.pointClouds[item.index][sensorId] = geometry
-
-            Session.dispatch(loadItem(item.index, sensorId))
-          },
+        attemptsMap[sensorId] = 0
+        const onLoad = (geometry: THREE.BufferGeometry) => {
+          Session.pointClouds[item.index][sensorId] = geometry
 
-          () => null,
-
-          () => {
+          Session.dispatch(loadItem(item.index, sensorId))
+        }
+        const onError = () => {
```

Comment:
Can we make finer error processing based on the ErrorEvent? https://github.com/mrdoob/three.js/blob/dev/src/loaders/FileLoader.d.ts#L17 Can we have the same error handling procedures for both images and point cloud?

Reply:
Ok give me a few days to dig in to how the `ErrorEvent` fields are populated

Reply:
From what I am seeing, the error is a `ProgressEvent`. It gives the status of the request. Is there any particular type of handling you want to check?
![image](https://user-images.githubusercontent.com/7757242/74287620-38a55a00-4cdf-11ea-9749-f5de43850652.png)

Reply:
Are all the errors supposed to be tried repeatedly?

Reply:
I think that for now we can just retry for all the errors. For point cloud, file formatting errors will not be caught in this section. I believe this is only for network errors. From what I have seen in the past, AWS will throw 404's sometimes even though the file exists. I can make another PR for the image loading later, but I would like this to be merged first for Dalu's use.